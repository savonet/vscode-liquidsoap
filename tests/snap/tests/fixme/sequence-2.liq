# Testing that head_ready is used in sequence#is_ready.
#
# We setup a sequence to start playing a source s, then stop playing the
# sequence, make s unavailable and start playing the sequence again.
# At this point the sequence should end its track and cleanup s *before*
# becoming unavailable.
#
# Pitfalls when cooking up this test:
#  - Does not work with a track insensitive switch to "instantly kill" the
#    source, because the switch makes sure to stay ready until the end of
#    track.
#  - Also does not work simply with a sequence underneath a fallible output,
#    because such an output will stop as soon as its source is not ready.

%include "test.liq"

# Create a fallible source. We use the experimental operator source.dynamic()
# which is bit fragile for a test.

flag = ref true
thread.run(delay=1., { flag:=false })
on = sine()
off = fail()
s = source.dynamic({ if !flag then [on] else [off] end })

test = sequence([ s, fail() ])

def check()
  if source.is_up(s) then
    test.fail()
  else
    test.pass()
  end
  shutdown()
end

output.dummy(fallback([test,on_track(fun(_)->check(),sine())]))
