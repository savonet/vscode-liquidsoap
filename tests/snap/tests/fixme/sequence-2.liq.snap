># Testing that head_ready is used in sequence#is_ready.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>#
#^^ source.liquidsoap comment.line.number-sign.liquidsoap
># We setup a sequence to start playing a source s, then stop playing the
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># sequence, make s unavailable and start playing the sequence again.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># At this point the sequence should end its track and cleanup s *before*
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># becoming unavailable.
#^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>#
#^^ source.liquidsoap comment.line.number-sign.liquidsoap
># Pitfalls when cooking up this test:
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>#  - Does not work with a track insensitive switch to "instantly kill" the
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>#    source, because the switch makes sure to stay ready until the end of
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>#    track.
#^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>#  - Also does not work simply with a sequence underneath a fallible output,
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>#    because such an output will stop as soon as its source is not ready.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>
>%include "test.liq"
#^^^^^^^^^ source.liquidsoap
#         ^ source.liquidsoap string.quoted.double.liquidsoap
#          ^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                  ^ source.liquidsoap string.quoted.double.liquidsoap
>
># Create a fallible source. We use the experimental operator source.dynamic()
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># which is bit fragile for a test.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>
>flag = ref true
#^^^^^^^^^^^^^^^^ source.liquidsoap
>thread.run(delay=1., { flag:=false })
#^^^^^^^^^^^^^^^^^ source.liquidsoap
#                 ^ source.liquidsoap constant.numeric.integer
#                  ^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>on = sine()
#^^^^^^^^^^^^ source.liquidsoap
>off = fail()
#^^^^^^^^^^^^^ source.liquidsoap
>s = source.dynamic({ if !flag then [on] else [off] end })
#^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                     ^^ source.liquidsoap keyword.control.liquidsoap
#                       ^^^^^^^ source.liquidsoap
#                              ^^^^ source.liquidsoap keyword.control.liquidsoap
#                                  ^^^^^^ source.liquidsoap
#                                        ^^^^ source.liquidsoap keyword.control.liquidsoap
#                                            ^^^^^^^ source.liquidsoap
#                                                   ^^^ source.liquidsoap keyword.control.liquidsoap
#                                                      ^^^^ source.liquidsoap
>
>test = sequence([ s, fail() ])
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>
>def check()
#^^^ source.liquidsoap keyword.control.liquidsoap
#   ^^^^^^^^^ source.liquidsoap
>  if source.is_up(s) then
#^^ source.liquidsoap
#  ^^ source.liquidsoap keyword.control.liquidsoap
#    ^^^^^^^^^^^^^^^^^ source.liquidsoap
#                     ^^^^ source.liquidsoap keyword.control.liquidsoap
>    test.fail()
#^^^^^^^^^^^^^^^^ source.liquidsoap
>  else
#^^ source.liquidsoap
#  ^^^^ source.liquidsoap keyword.control.liquidsoap
>    test.pass()
#^^^^^^^^^^^^^^^^ source.liquidsoap
>  end
#^^ source.liquidsoap
#  ^^^ source.liquidsoap keyword.control.liquidsoap
>  shutdown()
#^^^^^^^^^^^^^ source.liquidsoap
>end
#^^^ source.liquidsoap keyword.control.liquidsoap
>
>output.dummy(fallback([test,on_track(fun(_)->check(),sine())]))
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                                     ^^^ source.liquidsoap keyword.control.liquidsoap
#                                        ^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>